compiler/src/dotty/tools/dotc/quoted/ToolboxImpl.scala

```diff
@@ -15,8 +15,17 @@ object ToolboxImpl {
 
     private[this] val driver: QuoteDriver = new QuoteDriver(appClassloader)
 
+    private[this] var running = false
+
     def run[T](exprBuilder: QuoteContext => Expr[T]): T = synchronized {
-      driver.run(exprBuilder, settings)
+      try {
+        if (running) // detected nested run
+          throw new scala.quoted.Toolbox.ToolboxAlreadyRunning()
+        running = true
+        driver.run(exprBuilder, settings)
+      } finally {
+        running = false
+      }
```

Comment:
The documentation on the method `run` already says nested `run` is not supported, while this fix seems to suggest that nesting is fine if different toolboxes are used.

As the usage of nesting violates the protocol, I think an improvement of the documentation might be enough.

Reply:
It is the other fix that catches issues of nesting with different toolboxes.